doc/luasnip.txt

```diff
@@ -3628,197 +3673,423 @@ GENERAL ~
-    if all the text between them was deleted.
-- `filetype_extend(filetype:string, extend_filetypes:table of string)`: Tells
-    LuaSnip that for a buffer with `ft=filetype`, snippets from `extend_filetypes`
-    should be searched as well. `extend_filetypes` is a Lua array (`{ft1, ft2,
-    ft3}`). `luasnip.filetype_extend("lua", {"c", "cpp"})` would search and expand
-    C and C++ snippets for Lua files.
-- `filetype_set(filetype:string, replace_filetypes:table of string)`: Similar to
-    `filetype_extend`, but where _append_ appended filetypes, _set_ sets them:
-    `filetype_set("lua", {"c"})` causes only c snippets to be expanded in Lua
-    files; Lua snippets aren’t even searched.
-- `cleanup()`: clears all snippets. Not useful for regular usage, only when
-    authoring and testing snippets.
-- `refresh_notify(ft:string)`: Triggers an `autocmd` that other plugins can hook
-    into to perform various cleanup for the refreshed filetype. Useful for
-    signaling that new snippets were added for the filetype `ft`.
-- `set_choice(indx:number)`: Changes to the `indx`th choice. If no `choiceNode`
-    is active, an error is thrown. If the active `choiceNode` doesn’t have an
-    `indx`th choice, an error is thrown.
-- `get_current_choices() -> string[]`: Returns a list of multiline-strings
-    (themselves lists, even if they have only one line), the `i`th string
-    corresponding to the `i`th choice of the currently active `choiceNode`. If no
-    `choiceNode` is active, an error is thrown.
-- `setup_snip_env()`: Adds the variables defined (during `setup`) in `snip_env`
-    to the callers environment.
-- `get_snip_env()`: Returns `snip_env`.
-- `jump_destination(direction)`: Returns the node the next jump in `direction`
-    (either -1 or 1, for backwards, forwards respectively) leads to, or `nil` if
-    the destination could not be determined (most likely because there is no node
-    that can be jumped to in the given direction, or there is no active node).
-- `activate_node(opts)`: Activate a node in any snippet. `opts` contains the
-    following options:
-    - `pos`, `{[1]: row, [2]: byte-column}?`: The position at which a node should
-        be activated. Defaults to the position of the cursor.
-    - `strict`, `bool?`: If set, throw an error if the node under the cursor can’t
-        be jumped into. If not set, fall back to any node of the snippet and enter
-        that instead.
-    - `select`, `bool?`: Whether the text inside the node should be selected.
-        Defaults to true.
+    - `jump_into_func?: fun(snip: LuaSnip.Snippet) -> LuaSnip.Node?`
+
+This function returns:
+
+- `LuaSnip.ExpandedSnippet` The snippet that was inserted into the buffer.
+
+
+expand(opts?): boolean                                    *luasnip-api-expand*
+
+Find a snippet whose trigger matches the text before the cursor and expand it.
+
+- `opts?: LuaSnip.Opts.Expand?` Subset of opts accepted by `snip_expand`. Valid
+    keys are:
```

Comment:
another example (for testing ;) )

Reply:
5be06e2a47f931dd5ec9978d1b6a9e5f98323f39